packages/engine/src/simulation/package/state/packages/behavior_execution/package.js

```diff
@@ -195,7 +195,12 @@ const run_task = (experiment, sim, task_message, group_state, group_context) =>
             }
             
             agent_state.set_dynamic_access(behavior.dyn_access);
-            behavior.fn(agent_state, agent_ctx);
+            try {
+                behavior.fn(agent_state, agent_ctx);
+            } catch (e) {
+                // TODO expose/return user error properly
```

Comment:
The user error should be returned from this `run_task` function in `{ "errors": [error] }` and retrieved in `run_task` in `runner/javascript/mod.rs`.